src/DurableTask.AzureStorage/Partitioning/TablePartitionManager.cs

```diff
@@ -407,7 +407,9 @@ public async Task<ReadTableReponse> ReadAndWriteTableAsync(bool isShuttingDown,
                             throw;
                         }
 
-                        if (claimedLease)
+                        // Worker should listen to the control queue if 1) worker just claimed the lease,
+                        // or 2) worker is already the lease owner but hasn't listened to the queue yet for some reason.
+                        if (claimedLease || renewedLease)
```

Comment:
I _think_ this is what we want:

```suggestion
                        // Ensure worker is listening to the control queue iff either:
                        // 1) worker just claimed the lease,
                        // 2) worker was already the owner in the partitions table and is not actively draining the queue. Note that during draining, we renew the lease but do not want to listen to new messages. Otherwise, we'll never finish draining our in-memory messages.
                        bool isRenewingToDrainQueue = renewedLease & response.IsDrainingPartition
                        if (claimedLease || !isRenewingToDrainQueue)
```